### ISSUE TITLE
Allowed bicycles on paths with access=no and sac_scale=hiking

### ISSUE THREAD

**Issue by szymone** [valhalla/valhalla#4073]
Hi, there is an issue in graph.lua script which causes setting invalid access for bicycles on paths with the following tags: access=no, highway=path, sac_scale=hiking, for example https://www.openstreetmap.org/way/380268150. Because of access=no, no routing should be allowed on this way, however the following code sets bicycle access to true.

>     if kv["bike_tag"] == nil then
>       if kv["sac_scale"] == "hiking" then
>         kv["bike_forward"] = "true"
>         kv["bike_tag"] = "true"
>       elseif kv["sac_scale"] then
>         kv["bike_forward"] = "false"
>       end
>     end

https://github.com/valhalla/valhalla/blob/6e0db89cefc4e8d1b43f9a3ccb5874a99c166ee3/lua/graph.lua#L974

Is there any particular reason for relating bicycle access to sac_scale? Can we just remove this code fragment?

Thanks!

**Comment by nilsnolde** [comment]
> Can we just remove this code fragment?

IMO yes. Looks very strange, and no comments.. WDYT @gknisely @dnesbitt61?

**Comment by dnesbitt61** [comment]
I remember using sac_scale to prevent bicycle access. If a way is tagged as difficult for hiking it is almost certainly impassable on a bike. But we shouldn't use sac_scale to enable bicycle access so this doesn't look right.

**Comment by gknisely** [comment]
I think at the time if sac_scale was present then we set bicycle access to true for mtb folks.  We could keep that if we want, but we should at least check for the access tag value in this piece of code.  What do folks think is the best option here?

**Comment by nilsnolde** [comment]
> then we set bicycle access to true for mtb folks.

True, `hiking` sac_scale makes sense for mtb. Then we should set `bike_network_mask` on that sac_scale, not the general bike access right?

**Comment by szymone** [comment]
sac_scale tags are mostly used on highway=path and bikes are allowed on paths by default, so I don't think it's necessary to enable bike access when sac_scale=hiking. Maybe it'd be a good option to disable bike access when sac_scale != hiking, however in my opinion this shouldn't be connected. Values for sac_scale tag are often overestimated and for example mountain_hiking is used where hiking is enough and then bike access would be set to false.

I think this should be simplified and this code fragment should be removed.

**Comment by hungerburg** [comment]
The Wiki documentation for sac_scale says, that foot=yes is implied. I would not go so far to use sac_scale as an access tag though. For that the connection is much too brittle -- even hiking paths sometimes get closed, e.g. due to hazards. After all, it just tags the difficulty hikers are to expect there. An explicit foot=yes required, in my opinion. Never though, sac_scale=* implies bicycle=yes.

**Comment by szymone** [comment]
https://github.com/valhalla/valhalla/pull/4781
